tower_cli/cli/transfer/send.py

```diff
@@ -947,6 +974,95 @@ def import_schedules(self, existing_object, new_schedules, asset_type):
             except TowerCLIError as e:
                 self.log_error("Failed to delete schedule {} : {}".format(schedule_name, e))
 
+    def import_roles(self, existing_asset, new_roles, asset_type):
+        existing_role_data = common.extract_roles(existing_asset)
+        existing_roles = existing_role_data['items']
+        existing_name_to_id = existing_role_data['existing_name_to_object_map']
+
+        if new_roles == existing_roles:
+            self.log_ok("All roles are up to date")
+            return
+
+        existing_roles_by_name = {}
+        for role in existing_roles:
+            existing_roles_by_name[role['name']] = role
```

Comment:
I'm getting a little confused about name uniqueness here. AFAIK the role names (from the API) are things like "Admin", so this pivoting of a dict would lose almost all of the roles in the system, only copying memberships of 1 role of each type, which there are only a handful of.

But I'm sure there's more complex stuff going on, so basic questions about what's happening:

 - have you tested with different resource types with the same name, like test that role memberships are copied for a project named "foobar" and also copied for an inventory named "foobar" as well, and that doesn't break any of your logic?
 - are memberships in the same role type for each item copied correctly? In that example, memberships for the admin role of the project are copied distinctly from the admin role of the inventory?